src/ui/components/Library/Team/View/TestRuns/suspense/TestsCache.ts

```diff
@@ -39,3 +49,13 @@ export const testsCache = createCache<
     return orderBy(processedTests, "failureRate", "desc");
   },
 });
+
+const snapToTwoMinuteWindow = (date: Date, windowSizeInMinutes = 2) => {
+  const millisecondsInMinute = 60 * 1000;
+  const totalMilliseconds = date.getTime();
+  const totalMinutes = totalMilliseconds / millisecondsInMinute;
+  const roundedMinutes = Math.round(totalMinutes / windowSizeInMinutes) * windowSizeInMinutes;
+  const newTotalMilliseconds = roundedMinutes * millisecondsInMinute;
+  const snappedDate = new Date(newTotalMilliseconds);
+  return snappedDate;
+};
```

Comment:
I'm don't like this implementation but open to suggestion - considering the limitation we don't have a good way to like Test.created_at.

Reply:
I don't understand. Nothing is using this function. Why does it exist?

Reply:
Oh it _is_ being used https://github.com/replayio/devtools/blob/6e98cfd52741251163bd7ff28d2325121876a82e/src/ui/components/Library/Team/View/TestRuns/suspense/TestsCache.ts#L29

Nevermind.

Reply:
Oh weird. I thought I greped the file before commenting. My bad.

Reply:
Could you explain why this function exists though?

Reply:
Since the endTime is continuously changing on every toggle and IntervalCache cannot be used, I tried to find the nearest 'minute' to the given endTime, so we can avoid the API call on every toggle for that minute.